### PR TITLE
Add backorder count to graphql query

### PR DESCRIPTION
**Pull request by Jade-GG**

### CODE REVIEW

File: src/Model/Resolver/Quote/Backorder.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        /** @var Item $cartItem */
+        $cartItem = $value['model'];
+
+        // If used, grab the first configuration of a configurable item and use that
+        $configurableItems = $cartItem['qty_options'] ?? [];
+        if ($configurableItems) {
+            return reset($configurableItems)['backorders'] ?? 0;
+        }
```
Comment (Jade-GG) [inline src/Model/Resolver/Quote/Backorder.php:34]: I kind of make an assumption here, but I can't find a situation where you wouldn't just grab the first one?

**Comment by indykoning** [comment]
This looks like stock sensitive information which is not something everyone wants publically available.
I think it's best to add a check wether the site owner wants this information public.

Take a look at https://github.com/rapidez/magento2-compadre/blob/master/src/Model/Resolver/Inventory/StockItem.php#L52
https://github.com/rapidez/magento2-compadre/blob/master/src/Model/Config/Source/Fields.php#L29

Also don't forget to raise the version number in the module.xml so you can check if you can actually request this field yet:
https://github.com/rapidez/core/blob/e24943d1658dc2ea3a6165778862d89e572d6cd4/resources/views/cart/queries/cart.graphql#L23

**Comment by Jade-GG** [comment]
> This looks like stock sensitive information which is not something everyone wants publically available. I think it's best to add a check wether the site owner wants this information public.
> 
> Take a look at https://github.com/rapidez/magento2-compadre/blob/master/src/Model/Resolver/Inventory/StockItem.php#L52 https://github.com/rapidez/magento2-compadre/blob/master/src/Model/Config/Source/Fields.php#L29
> 
> Also don't forget to raise the version number in the module.xml so you can check if you can actually request this field yet: https://github.com/rapidez/core/blob/e24943d1658dc2ea3a6165778862d89e572d6cd4/resources/views/cart/queries/cart.graphql#L23

In principle this will only happen when you have backorders enabled, as it'll always return 0 unless you have more in your cart than available (which isn't normally possible). However maybe it's a good idea to check the setting as well?

**Comment by Jade-GG** [comment]
The magento setting now properly gets checked to make sure displaying this is enabled---i.e. the option `Allow Qty Below 0 and Notify Customer` is selected on either the product itself, or in the store config and the product is set to follow this option.

**Comment by indykoning** [comment]
If it's already in the public API then it's fine like this, if it is not we should make it configurable wether someone wants it in graphql as it was probably left out with good reason.

I've seen website competitors scan for weirder things so some webshops would probably rather not have it available via a public api

File: src/Model/Config/Source/Fields.php [hunk 2]
```diff
@@ -32,6 +32,7 @@ public function getProductStockItemFields()
             ['value' => 'min_sale_qty', 'label' => __('stock_item -> min_sale_qty')],
             ['value' => 'max_sale_qty', 'label' => __('stock_item -> max_sale_qty')],
             ['value' => 'qty_increments', 'label' => __('stock_item -> qty_increments')],
+            ['value' => 'qty_backordered', 'label' => __('stock_item -> qty_backordered')],
```
Comment (indykoning) [inline src/Model/Config/Source/Fields.php:35]: I'd change the label so it indicates it is part of the cart_item, not stock_item

**Pull request merged**